### PR TITLE
`evaluate` method for `ReductionOp`

### PR DESCRIPTION
**Pull request by Priya2698**
This is a part of resolving Issue #670. Addresses https://github.com/NVIDIA/Fuser/pull/899#discussion_r1330908237.

### CODE REVIEW

**Comment by Priya2698** [comment]
!build

File: csrc/ir/nodes.cpp [hunk 1]
```diff
@@ -1334,6 +1334,44 @@ std::string ReductionOp::toInlineString(int indent_size) const {
...
+      return {at::sum(input, reduction_axes)};
+      break;
+    case BinaryOpType::Max:
+      NVF_CHECK(
+        reduction_axes.size() == 1, 
+        "Operator type: ", 
+        getReductionOpType(), 
+        " expects one reduction dimension. Found ",
+        reduction_axes.size(), " reduction dimensions.")
+      return {std::get<0>(at::max(input, reduction_axes[0]))};
```
Comment (zasdfgbnm) [inline csrc/ir/nodes.cpp:1363]: Should we use `at::amax`? https://pytorch.org/docs/stable/generated/torch.amax.html

File: csrc/ir/nodes.cpp [hunk 2]
```diff
@@ -1334,6 +1334,44 @@ std::string ReductionOp::toInlineString(int indent_size) const {
   NVF_CHECK(false, "Tensor op can not be printed inline");
 }
 
+std::vector<PolymorphicValue> ReductionOp::evaluate(
+    const ExpressionEvaluator& ee,
+    const std::vector<PolymorphicValue>& inputs) const {
+  const auto& input = inputs.at(0).as<at::Tensor>();
+  const auto output = out()->as<TensorView>();
+
+  NVF_ERROR(output->getRFactorDomain().size() == 0,
```
Comment (zasdfgbnm) [inline csrc/ir/nodes.cpp:1343]: Should we use this? https://github.com/NVIDIA/Fuser/blob/6d4abc04dc4939b10e4076d4f70101a818084d26/csrc/ir/interface_nodes.h#L156

File: csrc/ir/nodes.cpp [hunk 3]
```diff
@@ -1334,6 +1334,39 @@ std::string ReductionOp::toInlineString(int indent_size) const {
...
 }
 
+std::vector<PolymorphicValue> ReductionOp::evaluate(
+    const ExpressionEvaluator& ee,
+    const std::vector<PolymorphicValue>& inputs) const {
+  const auto& input = inputs.at(0).as<at::Tensor>();
+  const auto output = out()->as<TensorView>();
+
+  NVF_ERROR(
+      output->hasRFactor(),
```
Comment (zasdfgbnm) [inline csrc/ir/nodes.cpp:1344]:
```suggestion
      !output->hasRFactor(),
```

Comment (Priya2698) [reply csrc/ir/nodes.cpp]: Oh right, thanks for catching this.

**Comment by Priya2698** [comment]
!build

**Comment by xwang233** [comment]
There might be some builder issue. Let me restart the CI job.

**Review by zasdfgbnm — APPROVED** [review]

**Pull request merged**